File: src/main/java/com/dk/sort/selection_sort.py
# Selection Sort
def selection_sort(A):
    n = len(A)  # length
    for i in range(n - 1):
        least = i  # temporary value for saving index.
        for j in range(i + 1, n):  # inner for loop. to compare with i and other index.
            if A[j] < A[least]:  # comparator
                least = j  # Update least variable.

        # Change Index and Values
        A[i], A[least] = A[least], A[i]
        printStep(A, i + 1)


def printStep(arr, val):
    print("   Step  %2d = " % val, end='')
    print(arr)


if __name__ == '__main__':
    # Test Code.
    a = [1, 3, 5]
    print(a)
    a[1], a[2] = a[2], a[1]
    print(a)

    data = [5, 3, 8, 4, 9, 1, 6, 2, 7]
    print("Original : ", data)
    selection_sort(data)
    print("Selction Sort : ", data)

File: src/main/java/com/dk/sort/selection_sort.java
package com.dk.sort;

import java.util.Arrays;

public class selection_sort {

    public int[] selection_sort(int[] arr){

        for(int i=0; i < arr.length; i++){

            int least = i;
            for(int j = i+1; j < arr.length; j++) {

                // Check minimum value.
                if(arr[least] > arr[j]){
                    // Update least.
                    least = j;
                }
            }

            // Change old least and new least.
            int tmp = arr[least];
            arr[least] = arr[i];
            arr[i] = tmp;

            printStep(arr, i + 1);
        }
        return arr;
    }


    void printStep(int[] arr, int val) {
        System.out.println("     Step  " + val + " = " + Arrays.toString(arr));

    }

    public static void main(String[] args){

        int[] data = {5, 3, 8, 4, 9, 1, 6, 2, 7};
        System.out.println(" Original : " + Arrays.toString(data));
        int[] result_data = new selection_sort().selection_sort(data);
        System.out.println(" Selection Sort : " + Arrays.toString(result_data));


    }

}
